typescript/src/entity.ts

```diff
@@ -52,6 +52,10 @@ export class Entity {
   // maps json names like 'id' to information about that property
   protected propertiesMap: Map<string, PropertyInfo>;
 
+  public get isDirty(): boolean {
```

Comment:
\o/ this is so JAVA

Reply:
typescript has a lot of kind of unusual, if not advanced, type operators and other interesting stuff.

but when you look at the default way it encourages you to write code, yeah, they don't even do a great job of modelling prototypical inheritance -- it's a very classed based OO model.

Reply:
anyway, this is actually somewhat useful --- it's really better if this variable isn't set from the outside.